lib/krane/deprecated_deploy_task.rb

```diff
@@ -447,6 +446,33 @@ def deploy_resources(resources, prune: false, verify:, record_summary: true)
       end
     end
 
+    def replace_or_create_resource(resource, force: false)
+      args = if force
+        ["replace", "--force", "--cascade", "-f", resource.file_path]
+      else
+        ["replace", "-f", resource.file_path]
+      end
+
+      _, _, status = kubectl.run(*args, log_failure: false)
+      return [nil, status] if status.success?
```

Comment:
Do we need to check if this actually a `Not Found` error, as opposed to, say, a timeout?

Reply:
We weren't doing that before, but that's a great idea. FYI there's a `raise_if_not_found` option on `run` that could be helpful for that (specifically, it raises `Krane::Kubectl::ResourceNotFoundError`).

Reply:
Is what I've done in my most recent commit what you had in mind?